core/capabilities/webapi/target/connector_handler.go

```diff
@@ -0,0 +1,115 @@
+package target
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/webcapabilities"
+)
+
+var _ connector.GatewayConnectorHandler = &ConnectorHandler{}
+
+type ConnectorHandler struct {
+	gc            connector.GatewayConnector
+	lggr          logger.Logger
+	responseChs   map[string]chan *api.Message
```

Comment:
how big do we expect this map to get? creates a new entry for every request, we probably need a cleanup after each read

Reply:
makes sense. will make that change